programs/validator-history/src/state.rs

```diff
@@ -454,4 +549,53 @@ mod tests {
     fn test_validator_history_layout() {
         println!("{}", ValidatorHistoryEntry::type_layout());
     }
+
+    #[test]
+    fn test_epoch_range() {
+        // Add in 4 CircBuf entries, with epoch 0, 1, 2, 3
+        let mut circ_buf = CircBuf::default();
+        for i in 0..4 {
+            let entry = ValidatorHistoryEntry {
+                epoch: i,
+                ..ValidatorHistoryEntry::default()
+            };
+            circ_buf.push(entry);
+        }
+        // Test epoch range [0, 3]
+        let epoch_range = circ_buf.epoch_range(0, 3).unwrap();
+        assert_eq!(epoch_range.len(), 4);
+
+        // skip an epoch in the middle
+        circ_buf = CircBuf::default();
+        for i in 0..4 {
+            if i == 2 {
+                continue;
+            }
+            let entry = ValidatorHistoryEntry {
+                epoch: i,
+                ..ValidatorHistoryEntry::default()
+            };
+            circ_buf.push(entry);
+        }
+
+        // Test epoch range [0, 3]
+        let epoch_range = circ_buf.epoch_range(0, 3).unwrap();
+        assert_eq!(epoch_range.len(), 3);
+
+        // same start and end epoch
+        let epoch_range = circ_buf.epoch_range(0, 0).unwrap();
+        assert_eq!(epoch_range.len(), 1);
+
+        // Test start epoch out of range
+        let epoch_range = circ_buf.epoch_range(4, 3);
+        assert!(epoch_range.is_none());
+
+        // Test end epoch out of range
+        let epoch_range = circ_buf.epoch_range(0, 5);
```

Comment:
imo this should prolly return [0,1,2,3,4]? what if you don't have the most recent epoch?

Reply:
returns all existing epochs in range now